src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_cipher.c

```diff
@@ -75,6 +76,24 @@ ARGS_NON_NULL_ALL static jobject GetAlgorithmName(JNIEnv* env, CipherInfo* type)
     return make_java_string(env, type->name);
 }
 
+int32_t AndroidCryptoNative_CipherIsSupported(CipherInfo* type)
+{
+    abort_if_invalid_pointer_argument (type);
+
+    JNIEnv* env = GetJNIEnv();
+    jobject algName = GetAlgorithmName(env, type);
+    if (!algName)
+        return FAIL;
+
+    jobject cipher = (*env)->CallStaticObjectMethod(env, g_cipherClass, g_cipherGetInstanceMethod, algName);
```

Comment:
Why does this line appear different than - say - line 106?

Reply:
To `GRef` creates a global reference, meaning you want it to live until you call `ReleaseGRef`. Line 106 creates a cipher instance and then stashes it in the `CipherCtx`. From the [JNI docs](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/design.html#referencing_java_objects):

> The JNI divides object references used by the native code into two categories: local and global references. Local references are valid for the duration of a native method call, and are automatically freed after the native method returns. Global references remain valid until they are explicitly freed.

But in this particular case, we don't need the `Cipher` instance to live, we just care if it throws. Since we don't need the object instance, it doesn't need to get promoted to a global reference, it can stay as a local and we can remove the local reference.